### PR TITLE
Update component package module loading

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

In #1333, a new approach to loading was introduced, but only for protocol. Extend it to connections, contracts and skills.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/helpers/base.py [hunk 1]
```diff
@@ -174,77 +172,6 @@ def load_aea_package(configuration: ComponentConfiguration) -> None:
         spec.loader.exec_module(module)  # type: ignore
 
 
-def load_all_modules(
```
Comment (DavidMinarsch) [inline aea/helpers/base.py:177]: Interesting how much code we no longer need ...

Comment (marcofavorito) [reply aea/helpers/base.py:177]: a lot of it was just not used anymore, and it has not been removed.

Comment (marcofavorito) [reply aea/helpers/base.py:177]: moreover, `load_all_modules` was more generic, hence it needed other code to work.

Comment (marcofavorito) [reply aea/helpers/base.py:177]: and, btw, we are still far from the ideal loading mechanism. We still don't handle correctly the conflicting versions for packages of different AEA projects in the same process...

Comment (DavidMinarsch) [reply aea/helpers/base.py:177]: I think we decided we won't handle that at all. We have the requirement that there is one version per process; that's it. Totally ok as its also the case with any other python package.

**Pull request merged**